services/src/main/java/org/exoplatform/webconferencing/externalvisio/rest/ExternalVisioConnectorRest.java

```diff
@@ -0,0 +1,88 @@
+import org.exoplatform.webconferencing.externalvisio.rest.util.EntityBuilder;
+import org.exoplatform.webconferencing.externalvisio.service.ExternalVisioConnectorService;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+@Path("/v1/externalVisio")
+@Tag(name = "/v1/externalVisio", description = "Manages external visio connector")
+public class ExternalVisioConnectorRest implements ResourceContainer {
+
+  private static final Log                    LOG = ExoLogger.getLogger(ExternalVisioConnectorRest.class);
+
+  private final ExternalVisioConnectorService externalVisioConnectorService;
+
+  public ExternalVisioConnectorRest(ExternalVisioConnectorService externalVisioConnectorService) {
+    this.externalVisioConnectorService = externalVisioConnectorService;
+  }
+
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @Operation(summary = "Creates a new External visio connector", description = "Creates a new External visio connector", method = "POST")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "400", description = "Invalid query input"),
+      @ApiResponse(responseCode = "500", description = "Internal server error"), })
+  public Response createExternalVisioConnector(@RequestBody(description = "ExternalVisioConnector object to create", required = true)
+  ExternalVisioConnector externalVisioConnector) {
+    if (externalVisioConnector == null) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("externalVisioConnector object is mandatory").build();
+    }
+    try {
+      ExternalVisioConnector createdExternalVisioConnector =
+                                                           externalVisioConnectorService.createExternalVisioConnector(EntityBuilder.toEntity(externalVisioConnector));
+      return Response.ok().entity(createdExternalVisioConnector).build();
+    } catch (Exception e) {
+      LOG.warn("Error creating a ExternalVisioConnector", e);
+      return Response.serverError().entity(e.getMessage()).build();
+    }
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @Path("/connectors")
+  @Operation(summary = "Retrieves the list of external visio connectors", description = "Retrieves the list of external visio connectors for an authenticated user", method = "GET")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "500", description = "Internal server error"), })
+  public Response getExternalVisioConnectors() {
+    try {
+      List<ExternalVisioConnector> externalVisioConnectors = externalVisioConnectorService.getExternalVisioConnectors();
```

Comment:
Could you add a cache system based on Etag ?

Reply:
The aim of the etag, is to no compute the list if the etag is still valid.
In this case, you compute the list for each request before checking the etag, so there is no benefit.

After reflexion, keep it simple, and remove the etag part, as it seems there is not a lot of benefit here